tests/console/validate_ext4_fs.pm

```diff
@@ -0,0 +1,66 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate ext4 filesystem and system partitioning
+# Maintainer: Sofia Syrianidou <ssyrianidou@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+
+sub run {
+
+    select_console "root-console";
+
+#Validate that swap is created and /home is on seperate partition. 
+script_run("df -PH /");
+script_run("df -PH /home");
+save_screenshot;
+
+    assert_script_run("lsblk -n | grep 'SWAP'", fail_message => "/swap partition was not created.");
+    assert_script_run("lsblk -n | grep 'home'", fail_message => "/home is not on separate partition.");
+
+#Validate that the filesystem type is the expected one.
+ 
+    my $test_data  = get_test_suite_data();
+    my @partitions = @{$test_data->{partitions}};
+
+    foreach (@partitions) {
+
+#Validate that the filesystem type is the expected one.
+
+        record_info("Check fs $_->{mnt_point}", "Verify the '$_->{mnt_point}' partition filesystem is '$_->{fs_type}'");
+        my $fstype = script_output("df -PT $_->{mnt_point} | grep -v \"Filesystem\" | awk '{print \$2}'");
+        assert_equals($_->{fs_type}, $fstype, "File system on '$_->{mnt_point}' partition does not correspond to the expected one.");
+
+#Validate that the size of root and home partition is the expected one.
+
+        record_info("Check fs $_->{mnt_point}", "Verify the '$_->{mnt_point}' partition size is '$_->{size}'");
+        my $fs_size = script_output("df -PH $_->{mnt_point} | grep -v \"Filesystem\" | awk '{print \$2}'");
+        assert_equals($_->{size}, $fs_size, "Filesystem size on '$_->{mnt_point}' partition does not correspond to the expected one.");
+
+
+}
+
+    
+#Test the ability to read and write files on /home and on root partition.
+    assert_script_run ("touch /home/emptyfile", fail_message => 'Failure while writting in /home');
```

Comment:
"Writing" has only one "t". Also, why not "echo Ornithorynque > /home/notemptyfile" instead ? Then the cat command below would actually print something. Just suggesting though.

Reply:
Done